db/system_keyspace.cc

```diff
@@ -2647,10 +2648,120 @@ class clients_table : public streaming_virtual_table {
+            for (auto& gid: groups) {
+                decorated_gids.push_back(decorated_gid{gid, make_partition_key(gid), shard});
+            }
+        }
+
+        // Must return partitions in token order.
+        std::sort(decorated_gids.begin(), decorated_gids.end(), [less = dht::ring_position_less_comparator(*_s)]
+                (const decorated_gid& l, const decorated_gid& r) { return less(l.key, r.key); });
+
+        for (auto& [gid, dk, shard]: decorated_gids) {
+            if (!contains_key(qr.partition_range(), dk)) {
+                continue;
+            }
+
+            auto cfg_opt = co_await _raft_gr.invoke_on(shard,
+                    [gid=gid] (service::raft_group_registry& raft_gr) -> std::optional<raft::configuration> {
+                // Be ready for a group to disappear while we're querying.
+                auto* srv = raft_gr.find_server(gid);
+                if (!srv) {
+                    return std::nullopt;
+                }
+                // FIXME: the configuration returned here is obtained from raft::fsm, it may not be
+                // persisted yet, so this is not 100% correct. It may happen that we crash after
+                // a config entry is appended in-memory in fsm but before it's persisted. It would be
+                // incorrect to return the configuration observed during this window - after restart
+                // the configuration would revert to the previous one.  Perhaps this is unlikely to
+                // happen in practice, but for correctness we should add a way of querying the
+                // latest persisted configuration.
+                return srv->get_configuration();
+            });
+
+            if (!cfg_opt) {
+                continue;
+            }
+
+            co_await result.emit_partition_start(dk);
+
+            // List current config first, because 'C' < 'P' and the disposition
+            // (ascii_type, 'CURRENT' vs 'PREVIOUS') is the first column in the clustering key.
+            co_await emit_member_set(result, "CURRENT", cfg_opt->current);
+            co_await emit_member_set(result, "PREVIOUS", cfg_opt->previous);
+
+            co_await result.emit_partition_end();
+        }
+    }
+
+private:
+    static schema_ptr build_schema() {
+        auto id = generate_legacy_id(system_keyspace::NAME, "raft_config");
+        return schema_builder(system_keyspace::NAME, "raft_config", std::make_optional(id))
+            .with_column("group_id", timeuuid_type, column_kind::partition_key)
+            .with_column("disposition", ascii_type, column_kind::clustering_key) // can be 'CURRENT` or `PREVIOUS'
+            .with_column("server_id", uuid_type, column_kind::clustering_key)
+            .with_column("can_vote", boolean_type)
+            .set_comment("Currently operating RAFT configuration")
+            .with_version(system_keyspace::generate_schema_version(id))
+            .build();
+    }
```

Comment:
As suggested in https://github.com/scylladb/scylladb/issues/12008 we will want to include more info (such as current term, leader etc) info this table. We may skip it for now but we have to have a plan how we will do it later. So what is the plan? Static columns?

Reply:
I was thinking static column too. Current term and leader need only a single row.
Also maybe `raft_config` is not an appropriate name if we want to include this info in the future (current term and leader are not part of the configuration if we want to use precise terminology). We could use something more generic like `raft_group_info`. Although using `raft_config` is not that a big problem.

Reply:
For someone who does not know raft internals `raft_config` should not be confusing, though may be `raft_state` would be better.

Reply:
"state" is on the other hand too generic to me, arguably the raft log is also part of the state (for example), but it's kept in a separate table.

Meh, let's just keep it as `raft_config`.

Reply:
Raft log *is* part of the state and one of the things I want to add to the table is its length in entries and bytes :)

Those other tables have different purpose. They are persisting the state of the raft needed for it to work. They are not for users to look at and may eventually be removed in favor of persisting directly in the commitlog.

I do not insist on the name though.

Reply:
Renamed to `raft_state` after all.